lib/bolt/pal.rb

```diff
@@ -144,7 +144,8 @@ def with_bolt_executor(executor, inventory, pdb_client = nil, applicator = nil,
           pdb_client,
           @hiera_config,
           @max_compiles
-        )
+        ),
+        squelch_parse_errors: true
```

Comment:
Following the stack trace in the ticket description at https://tickets.puppetlabs.com/browse/BOLT-100 I think that the `squelch_parse_errors` override option might have prevented it. I have a suspicion that while parsing some manifest code that assumed there was a `Catalog` which is not present in script mode (https://tickets.puppetlabs.com/browse/BOLT-1433?focusedCommentId=673243&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-673243). I am still working to reproduce this though.

Reply:
Steps to try to repro:
With the following manifest at `/test/manifests/init.pp` 
```puppet
class test {
  # execute 'apt-get update'
  exec { 'apt-update':                    # exec resource named 'apt-update'
    command => '/usr/bin/apt-get update'  # command this resource will run
  }

  # install apache2 package
  package { 'apache2':
    require => Exec['apt-update'],        # require 'apt-update' before installing
    ensure => installed,
  }

  # ensure apache2 service is running
  service { 'apache2':
    ensure => running,
  }

  # install mysql-server package
  package { 'mysql-server':
    require => Exec['apt-update'],        # require 'apt-update' before installing
    ensure => installed,
  }

  # ensure mysql service is running
  service { 'mysql':
    ensure => running,
  }

  # install php5 package
  package { 'php7.0':
    require => Exec['apt-update'],        # require 'apt-update' before installing
    ensure => installed,
  }

  # ensure info.php file exists
  file { '/var/www/html/info.php':
    ensure => file,
    content => '<?php  phpinfo(); ?>',    # phpinfo code
    require => Package['apache2'],        # require 'apache2' package before creating
  } 
}
```
I can run the following plan that successfully sets up the lamp stack both with and without the squelch_parse_errors override. 
```puppet
 plan test(TargetSpec $nodes){
  $nodes.get_resources('User')
  run_task('facts', $nodes)
  apply($nodes){
    include test
  }
  return $nodes.get_resources(['user', 'File[/tmp]'])
}
```
So far the only somewhat promising difference I've been able to come up with between using the override vs not is that without it I can trigger an error that results in an exception with
```
From: /home/cas/working_dir/bolt/bolt-modules/boltlib/lib/puppet/functions/run_plan.rb @ line 73 run_plan#run_plan:

    68:     # require modules to have dependencies defined in meta data.
    69:     loader = loaders.private_environment_loader
    70:     # Load resource types (full path commented below, TODO: delete)
    71:     # closure_scope.compiler.loaders.static_loader.runtime_3_init
    72:     # loaders.static_loader.runtime_3_init
 => 73:     require 'pry'; binding.pry
    74:     # TODO: Why would we not have a private_environment_loader?
    75:     unless loader && (func = loader.load(:plan, plan_name))
    76:       raise Puppet::ParseErrorWithIssue.from_issue_and_stack(
    77:         Puppet::Pops::Issues.issue(:UNKNOWN_PLAN) { Bolt::Error.unknown_plan(plan_name) }
    78:       )

[1] pry(#<#<Class:0x0000565028400af0>>)> closure_scope.environment.known_resource_types.loader.import_all
A 'define' expression is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/modules/concat/manifests/init.pp, line: 53, column: 1)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/modules/concat/manifests/init.pp, line: 94, column: 5)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/modules/concat/manifests/init.pp, line: 116, column: 7)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/modules/concat/manifests/init.pp, line: 124, column: 5)
```
Which can be avoided (and shows the warnings below pointing to the init.pp) when the `squelch_parse_error` symbol is set. 
```
 Host Class Definition is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 1, column: 1)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 3, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 8, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 14, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 19, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 25, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 30, column: 2)
A Resource Statement is only available when compiling a catalog (file: /home/cas/working_dir/bolt/Boltdir/site/test/manifests/init.pp, line: 36, column: 2)
```

Reply:
Should not need this setting using the generate face approach.